drupal/sites/default/boinc/modules/boincuser/boincuser.module

```diff
@@ -501,7 +501,7 @@ function boincuser_user($op, &$edit, &$account, $category = NULL) {
           }
 
           // Change boinc username
-          if ($edit['boincuser_name'] != $boinc_user->name) {
+          if ( $edit['boincuser_name'] and ($edit['boincuser_name'] != $boinc_user->name)) {
```

Comment:
The added SPACE at the front should be removed. Thanks.

Reply:
Latest commit, 79cd8bc, should fix this.